src/Arcus.Messaging.Pumps.Abstractions/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -1,16 +1,196 @@
 ﻿using System;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions;
 using Arcus.Messaging.Pumps.Abstractions.MessageHandling;
 using GuardNet;
-using Microsoft.Extensions.DependencyInjection;
 
-namespace Arcus.Messaging.Pumps.Abstractions.Extensions
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.DependencyInjection
 {
     /// <summary>
-    /// Extensions on the <see cref="IServiceCollection"/> related to message pumps.
+    /// Extensions on the <see cref="IServiceCollection"/> to add an general <see cref="IMessageHandler{TMessage,TMessageContext}"/> implementation.
     /// </summary>
     // ReSharper disable once InconsistentNaming
     public static class IServiceCollectionExtensions
     {
+        /// <summary>
+        /// Adds a <see cref="IMessageHandler{TMessage}" /> implementation to process the messages from an <see cref="MessagePump"/> implementation.
+        /// resources.
+        /// </summary>
+        /// <typeparam name="TMessageHandler">The type of the implementation.</typeparam>
+        /// <typeparam name="TMessage">The type of the message that the message handler will process.</typeparam>
+        /// <param name="services">The collection of services to use in the application.</param>
+        public static IServiceCollection WithMessageHandler<TMessageHandler, TMessage>(this IServiceCollection services)
```

Comment:
Do we want to have these general registrations? What is your goal?

Reply:
To have independent message determination and message handling. A serializer can be reused across message pumps to have a canonical format for different message handlers.

Reply:
Resolving as it's changed.